broadlink/climate.py

```diff
@@ -238,3 +238,324 @@ def set_schedule(self, weekday: List[dict], weekend: List[dict]) -> None:
+        data['cmnd_18'] = payload[0x18]
+
+        checksum = self._calculate_checksum(payload[:0x19]) # checksum=(payload[0x1a] << 8) + payload[0x19]
+
+        if (payload[0x19] == checksum[0] and payload[0x1a] == checksum[1]):
+            pass # success
+        else:
+            print('checksum fail', ['{:02x}'.format(x) for x in checksum])
+
+        if (payload_debug):
+            data['received_payload'] = payload
+
+        return data
+
+    def get_ac_info(self, payload_debug: bool = None) -> dict:
+        """Returns dictionary with A/C info...
+        Not implemented yet, except power state.
+
+        Args:
+            payload_debug (Optional[bool]): add the received payload for debugging
+        """
+        payload = self._send_short_payload(0)
+
+        # first 13 bytes are the same: 22 00 bb 00 07 00 00 00 18 00 01 21 c0
+        data = {}
+        data['state'] = True if (payload[0x0d] & 0b1 == 0b1) else False
+
+        if (payload_debug):
+            data['received_payload'] = payload
+
+        return data
+
+    def set_advanced(self,
+        state: bool,
+        mode: str,
+        target_temp: float,
+        speed: str,
+        swing_v: str,
+        swing_h: str,
+        sleep: bool,
+        display: bool,
+        health: bool,
+        cmnd_0d_rmask: int,
+        cmnd_0e_rmask: int,
+        cmnd_18: int,
+        checksum_lbit: int
+    ) -> bytes:
+        """Set paramaters of unit and return response. All parameters need to be specified.
+        
+        Args:
+            state (bool): power
+            target_temp (float): temperature set point 16<n<32
+            mode (str): cooling, heating, fan, dry, auto
+            speed (str): mute, low, mid, high, turbo (available only for cooling) 
+            swing_h (str): ON, OFF
+            swing_v (str): ON, OFF, 1, 2, 3, 4, 5 (fixed positions)
+            sleep (bool) 
+            display (bool)
+            health (bool)
+            cmnd_0d_rmask (int): unknown
+            cmnd_0e_rmask (int): unknown
+            cmnd_18 (int): unknown
+            checksum_lbit (int): subtracted from the left byte of the checksum
+        """
+        
+        PREFIX = [0x19, 0x00, 0xbb, 0x00, 0x06, 0x80, 0x00, 0x00, 0x0f, 0x00, 0x01, 0x01] # 12B
+        MIDDLE = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] # 13B + 2B checksum
+        SUFFIX = [0, 0, 0, 0, 0] # 5B
+        payload = PREFIX + MIDDLE + SUFFIX
+
+        assert ((target_temp >= 16) and (target_temp <= 32) and ((target_temp * 2) % 1 == 0))
+
+        if (swing_v == 'OFF'):
+            swing_L = 0b111
+        elif (swing_v == 'ON'):
+            swing_L = 0b000
+        else:
+            raise ValueError('unrecognized swing vertical value {}'.format(swing_v))
+
+        if (swing_h == 'OFF'):
+            swing_R = 0b111
+        elif (swing_h == 'ON'):
+            swing_R = 0b000
+        elif (swing_h >= 0 and swing_v <= 5):
+            swing_R = str(swing_v)
```

Comment:
swing_h and swing_v are strings. You need to cast them for integer comparisons.

Just out of curiosity... Does swing_R = '0b110' do anything? And are you sure you are not mixing swing_h and swing_v in this part?

Reply:
Added casting.
You're right I mixed which swing axis had the fixed positions.
`swing_R = '0b110'` gives a -5 error, since these positions are only available for the vertical.
This seems to have been the cause for the majority of the  issues I was having. Because the error is identical I thought it was an issue with the checksum, but it doesn't seem so.